mio/models/stream.py

```diff
@@ -82,48 +116,61 @@ class StreamBufferHeaderFormat(BufferHeaderFormat):
     battery_voltage_raw: int
     input_voltage_raw: int
 
+    _adc_scale: ADCScaling
+
 
 class StreamBufferHeader(BufferHeader):
     """
     Refinements of :class:`.BufferHeader` for
     :class:`~mio.stream_daq.StreamDaq`
+
+    .. todo::
+        Get the scaling factors from the device configuration
     """
 
     pixel_count: int
     battery_voltage_raw: int
     input_voltage_raw: int
-    _adc_scaling: ADCScaling = None
-
-    @property
-    def adc_scaling(self) -> Optional[ADCScaling]:
-        """
-        :class:`.ADCScaling` applied to voltage readings
-        """
-        return self._adc_scaling
 
-    @adc_scaling.setter
-    def adc_scaling(self, scaling: ADCScaling) -> None:
-        self._adc_scaling = scaling
+    _adc_scale: ADCScaling = PrivateAttr()
+    _battery_voltage_scaling: ScaledValue = PrivateAttr()
+    _input_voltage_scaling: ScaledValue = PrivateAttr()
+
+    def __init__(self, adc_scale: ADCScaling, **data: dict):
+        super().__init__(**data)
+        self._adc_scale = adc_scale
+        self._battery_voltage_scaling = ScaledValue(
+            scaling_factor=(
+                1
+                / (2**self._adc_scale.bitdepth)
+                * self._adc_scale.ref_voltage
+                * self._adc_scale.battery_div_factor
+            ),
+            maximum=self._adc_scale.battery_max_voltage,
+        )
+        self._input_voltage_scaling = ScaledValue(
+            scaling_factor=(
+                1
+                / (2**self._adc_scale.bitdepth)
+                * self._adc_scale.ref_voltage
+                * self._adc_scale.vin_div_factor
+            ),
+            maximum=self._adc_scale.vin_max_voltage,
+        )
```

Comment:
[Notes for whenever I can come back] Doing this for every header input feels like a waste and makes the header-filling process inconsistent. So, it is probably better to define a helper class that gets StreamBufferHeader and returns scaled values or something that the scaling factor is set once with the device config.